src/runtime/virtcontainers/clh_test.go

```diff
@@ -127,6 +127,31 @@ func TestCloudHypervisorAddVSock(t *testing.T) {
 	assert.Equal(clh.vmconfig.Vsock.Socket, "path")
 }
 
+func TestCloudHypervisorAddVfioDevice(t *testing.T) {
+	Id := "vfio123"
+	path := "/dev/vfio/101"
+
+	vfDevice := config.VFIODev{
+		ID:       Id,
+		SysfsDev: path,
+	}
+	testCloudHypervisorAddDevice(t, vfDevice)
+}
+
+func testCloudHypervisorAddDevice(t *testing.T, structure interface{}) {
+	assert := assert.New(t)
+	clh := cloudHypervisor{}
+	switch s := structure.(type) {
+	case config.VFIODev:
+		Id := "vfio123"
+		path := "/dev/vfio/101"
+		_ = clh.addVfioDevice(s)
+
+		assert.Equal(s.ID, Id)
+		assert.Equal(s.SysfsDev, path)
+	}
+}
+
```

Comment:
I don't think the abstraction of `testCloudHypervisorAddDevice` here is making sense. This function should be removed and everything can be done from `TestCloudHypervisorAddVfioDevice`.